put returns the result of the dispatch

funkjunky opened the pull request:
I needed the return of the put effect, because my middleware returns a selector if meta.selector is defined.

This seemed like an obvious addition that would only improve the library.

On top of that, I realized I neglected to update the src code last time. I'm not sure what parameters you use to compile the code, but I simply manually edited the src files to be correct.

wzrdzl commented:
@funkjunky thanks for the PR, could you please rebase your PR on current master, I added tests to the project and `commitlint` to make it easier to maintain this library. It would be super awesome (yet optional) if you add a test for the `put` effect or at least for the functionality you are adding. The example of the test can be found here:  https://github.com/wzrdzl/redux-yield-effect/blob/master/src/effects/call.spec.js

funkjunky commented:
Jesus, you installed the kitchen sink eh?

I added a test for put and added the missing library to package.json. Tests are passing.

With the specificity of your tests, any changes to your code would require an equal change to the test rendering them unnecessary.

It's also quite a bother to have to go through the commit message prompt for every commit.
I should have made three commits, but I didn't want to go through that prompt three times.

wzrdzl commented:
@funkjunky yeah it all looks a little bit heavy, but it helps to automate a lot of the maintenance burden. The standardised commit messages if done right help to automatically generate changelog and bump the right version number before releasing. Although I think we can remove commit message restrictions from the git hook and I can edit the message when merging the PR indeed, so thanks for the feedback!

About the specificity of the tests, I am really open to any proposals you think might help to make them more useful. I do not expect the lib code to change that significantly so changing test becomes an issue, but time will show.

wzrdzl commented:
@funkjunky I merged your branch with CLI ([the corresponding commit](https://github.com/wzrdzl/redux-yield-effect/commit/eacf6dd4724cf96e28161ba3e30ad5c6f73955a6)), don't know why GitHub didn't update the status of this PR.

Thank you anyways, the creds are yours :)

funkjunky commented:
Sweet, thanks. I'll try and find time to add more tests later. Cheers.
